diff --git a/src/test/fuzz/FuzzedDataProvider.h b/src/test/fuzz/FuzzedDataProvider.h
index 5fab0c4..5294102 100644
--- a/src/test/fuzz/FuzzedDataProvider.h
+++ b/src/test/fuzz/FuzzedDataProvider.h
@@ -23,2 +23,3 @@
 #include <initializer_list>
+#include <iostream>
 #include <limits>
@@ -143,2 +144,6 @@ inline std::string FuzzedDataProvider::ConsumeBytesAsString(size_t num_bytes) {
       reinterpret_cast<const std::string::value_type *>(data_ptr_), num_bytes);
+  std::cout << std::hex << "ifdp.push_bytes(&[";
+  for (uint8_t c : result)
+      std::cout << "0x" << unsigned(c) << ",";
+  std::cout << std::dec << "]); // (len=" << num_bytes << ")\n";
   Advance(num_bytes);
@@ -160,2 +165,3 @@ FuzzedDataProvider::ConsumeRandomLengthString(size_t max_length) {
   std::string result;
+  bool terminated = false;
 
@@ -169,4 +175,6 @@ FuzzedDataProvider::ConsumeRandomLengthString(size_t max_length) {
       Advance(1);
-      if (next != '\\')
+      if (next != '\\') {
+        terminated = true;
         break;
+      }
     }
@@ -176,2 +184,6 @@ FuzzedDataProvider::ConsumeRandomLengthString(size_t max_length) {
   result.shrink_to_fit();
+  std::cout << std::hex << "ifdp.push_str_u8(&[";
+  for (uint8_t c : result)
+      std::cout << "0x" << unsigned(c) << ",";
+  std::cout << std::dec << "]); // (len=" << result.size() << "), Limit: " << max_length << (terminated?"":"; not terminated! consider ifdp.push_str_u8_no_term instead?") << (result.size()==max_length?"Could be unlimited, if at the end?":"")<<"\n";
   return result;
@@ -233,2 +245,22 @@ T FuzzedDataProvider::ConsumeIntegralInRange(T min, T max) {
 
+  if (offset) {
+      std::cout << "ifdp.push_integral_in_range(";
+      if constexpr (std::is_signed_v<T>) {
+          std::cout << int64_t{static_cast<T>(static_cast<uint64_t>(min) + result)}
+                    << "i" << (sizeof(T) * 8) << ", "
+                    << int64_t{min}
+                    << "i" << (sizeof(T) * 8) << ", "
+                    << int64_t{max}
+                    << "i" << (sizeof(T) * 8) << ", ";
+      } else {
+          std::cout << uint64_t{static_cast<T>(static_cast<uint64_t>(min) + result)}
+                    << "u" << (sizeof(T) * 8) << ", "
+                    << uint64_t{min}
+                    << "u" << (sizeof(T) * 8) << ", "
+                    << uint64_t{max}
+                    << "u" << (sizeof(T) * 8) << ", ";
+      }
+      std::cout << ");\n";
+  }
+
   return static_cast<T>(static_cast<uint64_t>(min) + result);
@@ -251,2 +283,3 @@ T FuzzedDataProvider::ConsumeFloatingPointInRange(T min, T max) {
     abort();
+  std::cout << "ifdp.push_float(UNIMPLEMENTED);\n";
 
@@ -339,2 +372,6 @@ inline void FuzzedDataProvider::CopyAndAdvance(void *destination,
   std::memcpy(destination, data_ptr_, num_bytes);
+  std::cout << std::hex << "ifdp.push_bytes(&[";
+  for (uint8_t* c((uint8_t*)destination); c < (uint8_t*)destination + num_bytes; c++)
+      std::cout << "0x" << unsigned(*c) << ",";
+  std::cout << std::dec << "]); // (len=" << num_bytes << ")\n";
   Advance(num_bytes);

